webview/src/plots/components/templatePlots/TemplatePlotsGrid.tsx

```diff
@@ -90,63 +79,69 @@ export const TemplatePlotsGrid: React.FC<TemplatePlotsGridProps> = ({
     }
   }, [addDisabled, removeDisabled, disableClick])
 
-  const addEventsOnViewReady = () => {
+  const addEventsOnViewReady = useCallback(() => {
     const panels = document.querySelectorAll('.vega-bindings')
     for (const panel of Object.values(panels)) {
       panel.addEventListener('mouseenter', addDisabled)
       panel.addEventListener('mouseleave', removeDisabled)
       panel.addEventListener('click', disableClick)
     }
-  }
+  }, [addDisabled, removeDisabled, disableClick])
 
   const setEntriesOrder = (order: string[]) => {
     setOrder(order)
 
-    setSectionEntries(
-      groupIndex,
-      reorderObjectList<TemplatePlotEntry>(order, entries, 'id')
-    )
+    setSectionEntries(groupIndex, order)
   }
 
-  const reorderedItems = reorderObjectList<TemplatePlotEntry>(
-    order,
-    entries,
-    'id'
-  )
-
   const plotClassName = cx(styles.plot, {
     [styles.multiViewPlot]: multiView
   })
 
-  const items = reorderedItems.map((plot: TemplatePlotEntry) => {
-    const { id, content, multiView, revisions } = plot
-    const nbRevisions = (multiView && revisions?.length) || 1
-
-    const toggleDrag = (enabled: boolean) => {
+  const toggleDrag = useCallback(
+    (enabled: boolean, id: string) => {
       dispatch(changeDisabledDragIds(enabled ? [] : [id]))
-    }
+    },
+    [dispatch]
+  )
 
-    return (
-      <div
-        key={id}
-        className={plotClassName}
-        data-testid={`plot_${id}`}
-        id={id}
-        style={withScale(nbRevisions)}
-      >
-        <ZoomablePlot
-          id={id}
-          spec={{ ...content, ...autoSize } as VisualizationSpec}
-          onViewReady={addEventsOnViewReady}
-          toggleDrag={toggleDrag}
-          onResize={handleResize}
-          snapPoints={multiView ? [] : snapPoints}
-          currentSnapPoint={currentSize}
-          size={snapPoints[currentSize - 1]}
-        />
-      </div>
-    )
-  })
+  const items = useMemo(
+    () =>
+      order.map((plot: string) => {
+        const nbRevisions =
+          (multiView &&
+            plotDataStore[Section.TEMPLATE_PLOTS][plot].revisions?.length) ||
+          1
```

Comment:
[Q] Why set this to `|| 1`?

Reply:
I didn't change this `nbRevisions` tells the plot the number of column it should span, if it's multiview, it spans the number of revisions it has. If it's not multiview, we can say it contains one revision. I'll change the name of the variable to `colSpan` or something.